dags/analyzer_helper/discourse/extract_raw_info.py

```diff
@@ -0,0 +1,230 @@
+from datetime import datetime
+from analyzer_helper.discourse.utils.convert_date_time_formats import DateTimeFormatConverter
+from github.neo4j_storage.neo4j_connection import Neo4jConnection
+from typing import Optional
+
+
+class ExtractRawInfo:
+    def __init__(self, forum_endpoint:str):
+        """
+        Initialize the ExtractRawInfo with the forum endpoint and set up Neo4j connection.
+        """
+        self.neo4jConnection = Neo4jConnection()
+        self.driver = self.neo4jConnection.connect_neo4j()
+        self.forum_endpoint = forum_endpoint
+        self.converter = DateTimeFormatConverter()
+
+    def close(self):
+        """
+        Close the Neo4j connection.
+        """
+        self.driver.close()
+
+    def fetch_post_details(self, created_at: Optional[str] = None, comparison: Optional[str] = None) -> list:
+        """
+        Fetch details of posts from the Discourse forum.
+
+        :param created_at: Optional datetime string to filter posts created after this date.
+        :param comparison: Optional comparison operator, either 'gt' for greater than or 'gte' for greater than or equal to.
+        :return: List of dictionaries containing post details.
+        """
+        where_clause = ""
+        if created_at and comparison:
+            operator = '>' if comparison == 'gt' else '>='
+            where_clause = f"WHERE post.createdAt {operator} $createdAt"
```

Comment:
What's the reason to use `gte` rather than just using `gt` always?
Also, If you're going with both `gt` and `gte` please use an assertion to raise error if any other value was given.

Reply:
Reading the codes below, now I understand the why of `gt` and `gte` just the one thing is to make sure no other values would be given in function input. This could be achieved by doing an assertion